packages/codec/lib/format/utils/inspect.ts

```diff
@@ -0,0 +1,482 @@
+            return `Cannot decode indexed parameter of reference type ${
+              errorResult.error.type.typeClass
+            } (raw value ${errorResult.error.raw})`;
+          case "OverlongArraysAndStringsNotImplementedError":
+            return `Array or string is too long (length ${errorResult.error.lengthAsBN.toString()}); decoding is not supported`;
+          case "OverlargePointersNotImplementedError":
+            return `Pointer is too large (value ${errorResult.error.pointerAsBN.toString()}); decoding is not supported`;
+          case "UserDefinedTypeNotFoundError":
+          case "UnsupportedConstantError":
+          case "ReadErrorStack":
+          case "ReadErrorStorage":
+          case "ReadErrorBytes":
+            return Exception.message(errorResult.error); //yay, these five are already defined!
+        }
+      }
+    }
+  }
+}
+
+//these get their own class to deal with a minor complication
+class ContractInfoInspector {
+  value: Format.Values.ContractValueInfo;
+  constructor(value: Format.Values.ContractValueInfo) {
+    this.value = value;
+  }
+  [util.inspect.custom](depth: number | null, options: InspectOptions): string {
+    switch (this.value.kind) {
+      case "known":
+        return (
+          options.stylize(this.value.address, "number") +
+          ` (${this.value.class.typeName})`
+        );
+      case "unknown":
+        return (
+          options.stylize(this.value.address, "number") + " of unknown class"
+        );
+    }
+  }
+}
+
+function enumTypeName(enumType: Format.Types.EnumType) {
+  return (
+    (enumType.kind === "local" ? enumType.definingContractName + "." : "") +
+    enumType.typeName
+  );
+}
+
+function styleHexString(hex: string, options: InspectOptions): string {
+  return options.stylize(`hex'${hex.slice(2)}'`, "string");
+}
+
+//this function will be used in the future for displaying circular
+//structures
+function formatCircular(loopLength: number, options: InspectOptions): string {
+  return options.stylize(`[Circular (=up ${this.loopLength})]`, "special");
+}
+
+export function enumFullName(value: Format.Values.EnumValue): string {
+  switch (value.type.kind) {
+    case "local":
+      return `${value.type.definingContractName}.${value.type.typeName}.${
+        value.value.name
+      }`;
+    case "global":
+      return `${value.type.typeName}.${value.value.name}`;
+  }
+}
+
+//NOTE: Definitely do not use this in real code!  For tests only!
+//for convenience: invokes the nativize method on all the given variables
+export function nativizeVariables(variables: {
+  [name: string]: Format.Values.Result;
+}): { [name: string]: any } {
+  return Object.assign(
+    {},
+    ...Object.entries(variables).map(([name, value]) => ({
+      [name]: nativize(value)
+    }))
+  );
+}
+
+//HACK! Avoid using! Only use this if:
+//1. you absolutely have to, or
+//2. it's just testing, not real code
+export function nativize(result: Format.Values.Result): any {
```

Comment:
Huh, this is an interesting place to put these two functions.  I can see the logic, but do they really go here?  Maybe they should get their own file.

Reply:
I'm okay with them being in their own file, but I was trying to reduce the number of "buckets" users needed to be aware of... and since we don't want people to use `nativize`, I figured it shouldn't be so prominently visible.

Reply:
I mean, yeah, makes sense.